contract/contracts/NewInterfaces.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.6.0;
+
+contract DistributerInterface {
+    // TODO Add events
+
+    string name;
+    address tokenHolder;
+    mapping(address => address) tokenRefundDestinationList;
+
+    function balanceOf(address token) virtual external view returns (uint256) {}
+
+    function addToken() virtual external {}
+
+    function createCampaign(
+        address token,
+        string memory campaignInfoCid,
+        string memory recipientsCid,
+        string memory recipientsNum,
+        uint256 startDate
+    ) virtual external {}
+
+    // Future functionality
+    // function updateTokenHolder(address newTokenHolder) external; // onlyOwner
+}
+
+contract CampaignInterface {
+    // TODO Add events
+
+    string campaignInfoCid; // Contains campaign name and description as JSON
+    string recipientsCid; // Contains recipients value as JSON
+    uint32 recipientsNum;
+    uint32 claimedNum;
+    address refundDestination; // Use when campaign is cancelled
+
```

Comment:
How about setting `baseURL` as a variable and creating a function that returns https://ipfs.../.../{campaignInfoCid} with `campaignURI` ?